test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/PackageItemListViewModelTests.cs

```diff
@@ -37,5 +58,401 @@ public void LocalSources_PackageReader_NotNull()
             PackageReaderBase reader = func();
             Assert.IsType(typeof(PackageArchiveReader), reader);
         }
+
+        [Fact]
+        public void IconUrl_WithMalformedUrlScheme_ReturnsDefault()
+        {
+            var iconUrl = new Uri("httphttphttp://fake.com/image.png");
+
+            var packageItemListViewModel = new PackageItemListViewModel()
+            {
+                IconUrl = iconUrl
+            };
+
+            BitmapSource result = GetFinalIconBitmap(packageItemListViewModel).Result;
```

Comment:
Everywhere, instead of calling `GetFinalIconBitmap(...).Result` change the test's return type to `async Task` and this call to:
```C#
BitmapSource result = await GetFinalIconBitmapAsync(packageItemListViewModel);
```

Reply:
i'll try that again. when i tried it before, the Theory tests which had a bunch of test data instance would end up testing that method parallelly, which wasn't always happy.
let me try again and see how the tests like that now.

Reply:
fixed in next iteration. we'll see how the tests like it.